chart/templates/priorityclasses/priority-classes.yaml

```
@@ -0,0 +1,34 @@
+{{/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+*/}}
+
+#################################################
+## Priority classes provisioned via the chart values
+#################################################
+{{- range $e := .Values.priorityClasses }}
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: {{ .Release.Name }}-{{ $e.name }}
+{{- if $e.preemptionPolicy }}
+preemptionPolicy: {{ $e.preemptionPolicy }}
+{{ else }}
+preemptionPolicy: PreemptLowerPriority
+{{- end }}
+value: {{ $e.value }}
```

You need to use the required function if it's required or a default value if it is not:
```suggestion
value: {{ $e.value | required "value is required" }}
```

---

Good point. Added required clause as it is required and makes more sense than defining a default priority